stages/org.osbuild.cloud-init

```diff
@@ -142,12 +142,35 @@ SCHEMA = r"""
 """
 
 
+# Class representing the 'datasource_list' configuration option,
+# allowing to define a custom YAML dumper for it.
+class DatasourceList(list):
+    pass
+
+
+# Dedicated YAML dumper for the `DatasourceList` class.
+# Make sure that the `DatasourceList` always uses flow style.
+# https://cloudinit.readthedocs.io/en/latest/reference/base_config_reference.html#datasource-list
+# Specifically, it says for the `datasource_list` key:
+# "This key is unique in that it uses a subset of YAML syntax.
+#  It requires that the key and its contents, a list,
+#  must share a single line - no newlines."
+def datasourcelist_presenter(dumper, data):
+    return dumper.represent_sequence("tag:yaml.org,2002:seq", data, flow_style=True)
```

Comment:
I wasn't aware of this pyyaml feature, very nice! (TIL :)

Reply:
Me neither :sweat_smile: I had to do a bit of a research to determine how to do this...